### ISSUE TITLE
3.1-SNAPSHOT: ShadowNetwork failure on API 18

### ISSUE THREAD

**Issue by karlicoss** [robolectric/robolectric#2223]
Robolectric crashes with 

```
Caused by: java.lang.IllegalArgumentException: class org.robolectric.shadows.ShadowNetwork is not annotated with @Implements
    at org.robolectric.internal.bytecode.ShadowMap.getShadowInfo(ShadowMap.java:65)
    at org.robolectric.internal.bytecode.ShadowMap.get(ShadowMap.java:37)
    at org.robolectric.internal.bytecode.ShadowWrangler.getShadowConfig(ShadowWrangler.java:221)
    at org.robolectric.internal.bytecode.ShadowWrangler.findDirectShadowClass(ShadowWrangler.java:490)
    at org.robolectric.internal.bytecode.ShadowWrangler.classInitializing(ShadowWrangler.java:103)
    at org.robolectric.internal.bytecode.RobolectricInternals.classInitializing(RobolectricInternals.java:18)
    at android.net.Network.<clinit>(Network.java)
    ... 51 more
```

I guess this happens because of https://github.com/robolectric/robolectric/commit/dc56bf600028543cfc448e19816e389507d60248#diff-bfd93aa17d1d92f199b279b6610f22e5R42, there is no annotation on api < 21. I fixed it locally in my project by adding

```
@Implements(Network.class)
public class ShadowNetwork {
}
```

, but I guess it has to be fixed (@cermakcz ?).

**Comment by cermakcz** [comment]
I'm not sure if there should be the annotation, as there's no Network class on API < 21. I did it in a similar fashion as ShadowRenderNode is done.

**Comment by karlicoss** [comment]
Right, I see, other .vm files in Robolectric follow the same pattern.

So, I took a deeper look, a more detailed stacktrace:

```
Caused by: java.lang.ExceptionInInitializerError
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at com.squareup.okhttp.internal.Platform.findPlatform(Platform.java:136)
	at com.squareup.okhttp.internal.Platform.<clinit>(Platform.java:56)
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:137)
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:108)
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
	at com.squareup.okhttp.Call.getResponse(Call.java:286)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205)
	at com.squareup.okhttp.Call.execute(Call.java:80)
	at retrofit.client.OkClient.execute(OkClient.java:53)
	at com.yandex.mail.api.OkClientWithDbVersionHandler.execute(OkClientWithDbVersionHandler.java:53)
	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:326)
	... 33 more
Caused by: java.lang.IllegalArgumentException: class org.robolectric.shadows.ShadowNetwork is not annotated with @Implements
	at org.robolectric.internal.bytecode.ShadowMap.getShadowInfo(ShadowMap.java:65)
	at org.robolectric.internal.bytecode.ShadowMap.get(ShadowMap.java:37)
	at org.robolectric.internal.bytecode.ShadowWrangler.getShadowConfig(ShadowWrangler.java:221)
	at org.robolectric.internal.bytecode.ShadowWrangler.findDirectShadowClass(ShadowWrangler.java:490)
	at org.robolectric.internal.bytecode.ShadowWrangler.classInitializing(ShadowWrangler.java:103)
	at org.robolectric.internal.bytecode.RobolectricInternals.classInitializing(RobolectricInternals.java:18)
	at android.net.Network.<clinit>(Network.java)
	... 51 more
```

Apparently, it happens because of Okhttp, to be more precise, here https://github.com/square/okhttp/blob/47964b93fc48a41e25a09276bcd54b7292d3c851/okhttp/src/main/java/okhttp3/internal/Platform.java#L139 , so it does find the class in Robolectric and tries to load it. I wonder if we could come to some generic solution in Robolectric to fix this, although, right now, like I said, in my project, I fixed it by adding my stub annotated with `@Implements`.

**Comment by xian** [comment]
Fixed as of 3.1.3.

**Comment by alixwar** [comment]
I'm still seeing this issue (3.4) if I remove the workaround desribed in the original post
